gensim/models/keyedvectors.py

```diff
@@ -154,6 +154,66 @@ def get_vector(self, entity):
         else:
             raise KeyError("'%s' not in vocabulary" % entity)
 
+    def add(self, entities, weights, replace=False):
+        """Add entities and theirs vectors in a manual way.
+        If some entity is already in the vocabulary, old vector is keeped unless `replace` flag is True.
+
+        Parameters
+        ----------
+        entities : list of str
+            Entities specified by string tags.
+        weights: {list of numpy.ndarray, numpy.ndarray}
+            List of 1D np.array vectors or 2D np.array of vectors.
+        replace: bool, optional
+            Flag indicating whether to replace vectors for entities which are already in the vocabulary,
+            if True - replace vectors, otherwise - keep old vectors.
+        """
+        if isinstance(entities, string_types):
+            entities = [entities]
+            weights = weights.reshape(1, -1)
```

Comment:
probably, should be `weights = np.array(weights).reshape(1, -1)` for case if weights, for example, list of floats